tests/test__collection_api.py

```diff
@@ -4831,6 +4831,83 @@ def test__aggregate_to_string(self):
         }]
         self.assertEqual(expect, list(actual))
 
+    @skipIf(not _HAVE_PYMONGO, 'pymongo not installed')
+    def test__aggregate_to_decimal(self):
```

Comment:
Please also add such a test in test__mongomock to make sure the feature stays in sync with the actual pymongo behavior.

Reply:
I have add the test to this PR, thank you for such thorough code review.